File: query-feature.js
const sql = require('sql.js');
const fs = require('fs');

async function main() {
  const SQL = await sql.default();
  const buffer = fs.readFileSync('D:\\Programmi\\PORTFOLIO\\decision-intelligence-journal-2.0\\features.db');
  const db = new SQL.Database(buffer);
  const result = db.exec('SELECT * FROM features WHERE id = 245');
  if (result.length > 0) {
    const columns = result[0].columns;
    const values = result[0].values[0];
    const row = {};
    columns.forEach((col, i) => row[col] = values[i]);
    console.log(JSON.stringify(row, null, 2));
  }
  db.close();
}
main();

File: query_feature.py
import sqlite3

conn = sqlite3.connect(r'D:\Programmi\PORTFOLIO\decision-intelligence-journal-2.0\features.db')
cursor = conn.cursor()
cursor.execute('SELECT id, priority, category, name, description, steps, passes, in_progress FROM features WHERE id = 249')
row = cursor.fetchone()

if row:
    print(f"ID: {row[0]}")
    print(f"Priority: {row[1]}")
    print(f"Category: {row[2]}")
    print(f"Name: {row[3]}")
    print(f"Description: {row[4]}")
    print(f"Steps: {row[5]}")
    print(f"Passes: {row[6]}")
    print(f"In Progress: {row[7]}")
else:
    print("Feature 249 not found")

conn.close()
